opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -653,26 +656,25 @@ std::string CpGrid::name() const
 
 int CpGrid::maxLevel() const
 {
-    if (!distributed_data_.empty()){
-        return 0;
-    }
-    if (data_.size() == 1){
+    if (currentData().size() == 1){
         return 0; // "GLOBAL" grid is the unique one
     }
     else {  // There are multiple LGRs
-        return double(this -> data_.size() - 2); // last entry is leafView, and it starts in level 0 = GLOBAL grid.
+        return this -> currentData().size() - 2; // last entry is leafView, and it starts in level 0 = GLOBAL grid.
     }
 }
 
 template<int codim>
-typename CpGridTraits::template Codim<codim>::LevelIterator CpGrid::lbegin (int level) const{
+typename CpGridTraits::template Codim<codim>::LevelIterator CpGrid::lbegin (int level) const
+{
     if (level<0 || level>maxLevel())
         DUNE_THROW(GridError, "levelIndexSet of nonexisting level " << level << " requested!");
-    if (!distributed_data_.empty()){
-        return cpgrid::Iterator<codim, All_Partition>(*current_view_data_, 0, true);
+    bool areThereLgrs =  (currentData().size() > 1);
+    if(areThereLgrs) {
+        return  cpgrid::Iterator<codim, All_Partition>( *(*current_data_)[level], 0, true);
     }
-    else{
-        return cpgrid::Iterator<codim, All_Partition>(*data_[level], 0, true);
+    else {
+        return cpgrid::Iterator<codim, All_Partition>(*current_view_data_, 0, true);
```

Comment:
Do we really need if-else? Isn't the level grid view always in current_data_[level]?

Reply:
You're right. No need for this if-statement (I assume also for the ones you point out below). I'll remove them. Thanks!